gammapy/obs/observation.py

```diff
@@ -319,3 +339,569 @@ def select_observations(self, selection=None):
+        Array of values or bin edges, depending on the **format** parameter.
+    format : str
+        Format of binning specified: ``bin_edges``, ``bin_values``.
+
+    Examples
+    --------
+    Create a few axes:
+
+    .. code:: python
+
+        alt = Angle([0, 30, 60, 90], 'degree')
+        alt_obs_group_axis = ObservationGroupAxis('ALT', alt, 'bin_edges')
+        az = Angle([-90, 90, 270], 'degree')
+        az_obs_group_axis = ObservationGroupAxis('AZ', az, 'bin_edges')
+        ntels = np.array([3, 4])
+        ntels_obs_group_axis = ObservationGroupAxis('N_TELS', ntels, 'bin_values')
+    """
+
+    def __init__(self, name, bins, format):
+        if format not in ['bin_edges', 'bin_values']:
+            raise ValueError("Invalid bin format {}.".format(self.format))
+        self.name = name
+        self.bins = bins
+        self.format = format
+
+    @property
+    def n_bins(self):
+        """Number of bins (int)"""
+        if self.format == 'bin_edges':
+            return len(self.bins) - 1
+        elif self.format == 'bin_values':
+            return len(self.bins)
+
+    def get_bin(self, bin_id):
+        """Get bin (int, float or `~astropy.units.Quantity`-like)
+
+        Value or tuple of bin edges (depending on the **format** parameter)
+        for the specified bin.
+
+        Parameters
+        ----------
+        bin_id : int
+            ID of the bin to retrieve.
+
+        Returns
+        -------
+        bin : int, float or `~astropy.units.Quantity`-like
+            Value or tuple of bin edges, depending on the **format** parameter.
+        """
+        if self.format == 'bin_edges':
+            return (self.bins[bin_id], self.bins[bin_id + 1])
+        elif self.format == 'bin_values':
+            return self.bins[bin_id]
+
+    @property
+    def get_bins(self):
+        """List of bins (int, float or `~astropy.units.Quantity`-like)
+
+        List of bin edges or values (depending on the **format** parameter)
+        for all bins.
+        """
+        bins = []
+        for i_bin in np.arange(self.n_bins):
+            bins.append(self.get_bin(i_bin))
+        return bins
+
+    @classmethod
+    def from_column(cls, col):
+        """Import from astropy column.
+
+        Parameters
+        ----------
+        col : `~astropy.table.Column`
+            Column with the axis info.
+        """
+        return cls(name=col.name,
+                   bins=col,
+                   format=col.meta['axis_format'])
+
+    def to_column(self):
+        """Convert to astropy column.
+
+        Returns
+        -------
+        col : `~astropy.table.Column`
+            Column with the axis info.
+         """
+        col = Column(data=self.bins, name=self.name)
+        col.meta['axis_format'] = self.format
+        return col
+
+    @property
+    def info(self):
+        """Info string (str)"""
+        return ("{0} {1} {2}".format(self.name, self.format, self.bins))
+
+    def print(self):
```

Comment:
Remove?
(it's rare to want to do specifically that, and in those cases the caller can just use the one line in this method)